platform/platform_tests/distance_tests.cpp

```diff
@@ -282,19 +282,19 @@ UNIT_TEST(Distance_FormattedDistance)
     {Distance(991,     Units::Feet),       990,  Units::Feet,       "990",  "990 ft"},
 
     // From Miles to Miles
-    {Distance(0,       Units::Miles),      0,    Units::Feet,       "0",    "0 ft"},
-    {Distance(0.1,     Units::Miles),      530,  Units::Feet,       "530",  "530 ft"},
-    {Distance(1,       Units::Miles),      1.0,  Units::Miles,      "1.0",  "1.0 mi"},
-    {Distance(1.234,   Units::Miles),      1.2,  Units::Miles,      "1.2",  "1.2 mi"},
-    {Distance(9.99,    Units::Miles),      10,   Units::Miles,      "10",   "10 mi"},
-    {Distance(10.01,   Units::Miles),      10,   Units::Miles,      "10",   "10 mi"},
-    {Distance(11,      Units::Miles),      11,   Units::Miles,      "11",   "11 mi"},
-    {Distance(54,      Units::Miles),      54,   Units::Miles,      "54",   "54 mi"},
-    {Distance(145,     Units::Miles),      145,  Units::Miles,      "145",  "145 mi"},
-    {Distance(999,     Units::Miles),      999,  Units::Miles,      "999",  "999 mi"},
-    {Distance(1149.99, Units::Miles),      1150, Units::Miles,      "1150", "1150 mi"},
-    {Distance(1150,    Units::Miles),      1150, Units::Miles,      "1150", "1150 mi"},
-    {Distance(1150.01, Units::Miles),      1150, Units::Miles,      "1150", "1150 mi"},
+    {Distance(0,       Units::Miles),      0,    Units::Feet,       "0",     "0 ft"},
+    {Distance(0.1,     Units::Miles),      530,  Units::Feet,       "530",   "530 ft"},
+    {Distance(1,       Units::Miles),      1.0,  Units::Miles,      "1.0",   "1.0 mi"},
+    {Distance(1.234,   Units::Miles),      1.2,  Units::Miles,      "1.2",   "1.2 mi"},
+    {Distance(9.99,    Units::Miles),      10,   Units::Miles,      "10",    "10 mi"},
+    {Distance(10.01,   Units::Miles),      10,   Units::Miles,      "10",    "10 mi"},
+    {Distance(11,      Units::Miles),      11,   Units::Miles,      "11",    "11 mi"},
+    {Distance(54,      Units::Miles),      54,   Units::Miles,      "54",    "54 mi"},
+    {Distance(145,     Units::Miles),      145,  Units::Miles,      "145",   "145 mi"},
+    {Distance(999,     Units::Miles),      999,  Units::Miles,      "999",   "999 mi"},
+    {Distance(1149.99, Units::Miles),      1150, Units::Miles,      "1,150", "1,150 mi"},
+    {Distance(1150,    Units::Miles),      1150, Units::Miles,      "1,150", "1,150 mi"},
+    {Distance(1150.01, Units::Miles),      1150, Units::Miles,      "1,150", "1,150 mi"},
```

Comment:
1. Can tests for a decimal separator also be added?
2. Is it hard to add tests for altitude formatting?
3. What if tests are run on a French locale?

Reply:
> 1. Can tests for a decimal separator also be added?

Do you mean to test decimal separators on different locales? With the current implementation in `ToStringPrecision()`, `GetCurrentLocale()` is only called once at start-up, so we could only make tests with a single locale.

> 2. Is it hard to add tests for altitude formatting?

We already have a couple of tests for altitude formatting in `UNIT_TEST(Distance_CreateAltitudeFormatted)` in `distance_tests.cpp`. I've included tough a third one for high altitudes in meters.

> 3. What if tests are run on a French locale?

Current implementation of `GetCurrentLocale()` in linux does not read system's locale, and forces decimal and grouping separators to '.' and ',' respectively, so tests run flawlessly even with Spanish or French system locales.

Reply:
1/ I meant test cases where a decimal separator is used.